docs/content/concepts/assets/external-assets.mdx

```diff
@@ -1,73 +1,68 @@
 ---
-title: External Assets | Dagster
-description: External assets model assets in Dagster that are not scheduled or materialized in Dagster.
+title: Non-Materializable Assets | Dagster
```

Comment:
tl;dr

1) I think we should have class that represents a node in the asset graph. Call it `AssetNode`. (We could repurpose `AssetSpec` for this)

2) There should be "execution functions" that can cut across any number of nodes.

3) `@asset` is now framed "Short hand for declaring an asset node in the graph and an execution function that materializes it"

"Non-materializable" is a mouthful and overly specific IMO.

Reply:
We chatted about this in person. Summarizing my thoughts here for posterity:
- I entirely agree on points 1, 2, and 3. Some relevant thoughts in this discussion comment which I think are very in line with these points: https://github.com/dagster-io/internal/discussions/9371#discussioncomment-9190609.
- I still think we need a way to refer to assets that don't have a materialization function that Dagster can invoke. The way that users work with these assets in Dagster is often different than the way they work with materializable assets. And it's useful to have a docs page that covers the different ways to capture metadata for these assets and how we view their value.  What I like about "non-materializable asset" is that I think it implies the ontology that's implied by points 1, 2, and 3.